src/Css/Stylesheet.php

```diff
@@ -1609,7 +1609,7 @@ private function _parse_font_face($str)
      */
     private function _parse_properties($str)
     {
-        $properties = preg_split("/;(?=(?:[^\(]*\([^\)]*\))*(?![^\)]*\)))/", $str);
+        $properties = preg_split("/;(?=(?:[^\(]*\((.*)?[^\)]*\))*(?![^\)]*\)))/", $str);
```

Comment:
Is this change necessary? It looks like you're capturing the content of the potential CSS function but not doing anything with it?

Reply:
I am not a hero with regex, so I might have done it wrong. But I made chis change to get `var(var(...))` to work. Without this change all properties became one big property and everything broke. Arguably this is still not right because `var(var(var(...)))` might still fail, but I don't really know how realistic that is.

Reply:
Let me think on that, then. RegEx isn't great for parsing out nested patterns like that so it may be hard to fully accommodate.

Reply:
This does work, so good job. I believe you can exclude the capturing group since it's not being used:

```suggestion
        $properties = preg_split("/;(?=(?:[^\(]*\(.*?[^\)]*\))*(?![^\)]*\)))/", $str);
```

Reply:
Ah yes, of course. Good idea.

Reply:
Hm, from reading the pattern, it looks like it is supposed to detect a single closing parenthesis (with balanced parentheses in-between), so that a semicolon can be used in a `url()` function, e.g. `url(123;45.png)`; are there other cases where a semicolon makes sense, obviously outside a string? Based on that, I think it would be better to simplify the pattern radically (pattern with added white space for readability):

```
"/; (?! [^\(]* \) )/x"
```

So only try to detect a lone closing parenthesis after the semicolon, and don’t bother with trying to detect balanced pairs. I have prepared a PR based on that pattern.

Reply:
See #3323